src/Packetery/Module/Framework/WcCartTrait.php

```diff
@@ -72,4 +74,23 @@ public function cart(): ?\WC_Cart {
 	public function cartCalculateTotals(): void {
 		WC()->cart->calculate_totals();
 	}
+
+	/**
+	 * Value is cast to float because PHPDoc is not reliable.
+	 */
+	public function cartGetSubtotal(): float {
+		return (float) WC()->cart->get_subtotal();
+	}
+
+	public function cartGetCustomer(): WC_Customer {
+		return WC()->cart->get_customer();
+	}
+
+	public function cartCalculateShipping(): array {
+		return WC()->cart->calculate_shipping();
+	}
+
+	public function cartFeesApiAddFee( array $args ): void {
+		WC()->cart->fees_api()->add_fee( $args );
```

Comment:
add_fee vrací případné chyby, nebylo by vhodné to ošetřit?

Reply:
To že jsme se s tím nesetkali, tak neznemaná, že to nemohlo někdy nastat u klienta. Raději ať je to příliš defensivní než to pak řešit jako bug